response_operations_ui/views/accounts.py

```diff
@@ -40,6 +49,58 @@ def get_my_account():
         return jsonify(e)
 
 
+@account_bp.route("/my-account", methods=["POST"])
```

Comment:
I think this could be merged into the GET my-account above.  Instead of having to do a redirect on a fail, it'd be quicker, and look less weird if you were looking at the network traffic, if you put the POST code at the top of that function, and if it fails, you just flash an error then just do all the other stuff you'd normally do in the GET.